### ISSUE TITLE
Minimum C standard C89, C99, C11

### ISSUE THREAD

**Issue by NeroBurner** [FNA-XNA/FAudio#144]
What's the required minimum C-standard?

Found the following warning when enabling the -pedantic switch:
```
FAudio_internal.h:482: warning: ISO C99 doesn’t support unnamed structs/unions [-Wpedantic]
  };
   ^
```

anonymous unions are a C11 extension

**Comment by flibitijibibo** [comment]
It's supposed to be C99, but naming that union would be... quite the addition. I'd like to silence this one for Wpedantic if possible, since all supported compilers are okay with this.

**Comment by NeroBurner** [comment]
I'm trying to follow your feedback on the DumpVoices PR

> 3. General style fixes and C compliance issues, try to match the rest of the source and check against various compilers with strict rules (VS2010 is my go-to since it has super basic C support)

But I don't want to install VS2010 to test compliance. So I tried to add warning flags

**Comment by NeroBurner** [comment]
couldn't we just push the requirement to C11?

**Comment by flibitijibibo** [comment]
You can for your qtcreator project if you want, but honestly if the only thing pushing us to C11 is that the compiler doesn't like `union { };` I'm not really sure I care about Wpedantic in this case. If you really want, you can go figure out a way to incorporate something like `DUMMYUNIONNAME` and `NONAMELESSUNION` like Wine does, but even this will be a lot more work than you think it'll be.

**Comment by flibitijibibo** [comment]
Looked into this some more, would simply changing C_STANDARD to gnu99 resolve this?

**Comment by NeroBurner** [comment]
does MSVC support gnu extensions?

and I think it is not enough
```
/bin/cc -DFAUDIO_DISABLE_DEBUGCONFIGURATION -DFAudio_EXPORTS -DHAVE_FFMPEG=1 -I/home/nero/repos/FAudio/src -I/home/nero/repos/FAudio/include -I/home/nero/repos/FAudio/SYSTEM -isystem /usr/include/SDL2  -O3 -DNDEBUG -fPIC   -Wall -Wextra -pedantic -Wno-unused-parameter -Wno-unused-but-set-variable -std=gnu99 -o CMakeFiles/FAudio.dir/src/XNA_Song.c.o   -c /home/nero/repos/FAudio/src/XNA_Song.c
In file included from /home/nero/repos/FAudio/src/XNA_Song.c:29:
/home/nero/repos/FAudio/src/FAudio_internal.h:482:3: warning: ISO C99 doesn’t support unnamed structs/unions [-Wpedantic]
  };
```

**Comment by flibitijibibo** [comment]
It doesn’t, but it also supports anonymous unions by default.

**Comment by NeroBurner** [comment]
is there anything stopping us to just push the minimum required C version to C11?

**Comment by flibitijibibo** [comment]
We don’t use any actual C11 features. Turn this off via `#pragma GCC diagnostic push` if you have to, I really do not care about this warning.

**Comment by flibitijibibo** [comment]
Resolved by #146
